### PR TITLE
Fix vae_mnist model

### PR DESCRIPTION
**Pull request by joost-ve**
- Replaces CUDAapi with CUDA
- Replaces Flux.logitbinarycrossentropy with Flux.Losses.logitbinarycrossentropy
- Makes the Encoder a functor so will actually be moved to the CPU before saving the BSON file

see also https://github.com/FluxML/model-zoo/issues/266

### CODE REVIEW

File: vision/vae_mnist/vae_mnist.jl [hunk 1]
```diff
@@ -6,10 +6,11 @@
 
 using Base.Iterators: partition
 using BSON
-using CUDAapi: has_cuda_gpu
+using CUDA
 using DrWatson: struct2dict
 using Flux
-using Flux: logitbinarycrossentropy, chunk
+using Flux: @functor, chunk
```
Comment (CarloLucibello) [inline vision/vae_mnist/vae_mnist.jl:12]: looks like chunk is not used

File: vision/vae_mnist/vae_mnist.jl [hunk 2]
```diff
@@ -6,10 +6,11 @@
 
 using Base.Iterators: partition
```
Comment (CarloLucibello) [inline vision/vae_mnist/vae_mnist.jl:7]: partition not used

**Comment by CarloLucibello** [comment]
Is generation fine or producing grey images?

File: vision/vae_mnist/vae_mnist.jl (hunk 1 above)
Comment (joost-ve) [reply vision/vae_mnist/vae_mnist.jl:12]: It is used in the convert_to_image function: https://github.com/FluxML/model-zoo/pull/267/files#diff-d2fcc7489d82937b5b466b70d629d200f23003b00a0e923cfa5031eedfa54d4cL70

**Comment by joost-ve** [comment]
Just grey images

**Comment by joost-ve** [comment]
Runs without issues and resulting images are ok now. It could use some improvement overall, but I think we should just leave it like this for now.

File: vision/vae_mnist/vae_mnist.jl [hunk 3]
```diff
@@ -59,15 +61,15 @@ function model_loss(encoder, decoder, λ, x, device)
     # KL-divergence
     kl_q_p = 0.5f0 * sum(@. (exp(2f0 * logσ) + μ^2 -1f0 - 2f0 * logσ)) / len
 
-    logp_x_z = -sum(logitbinarycrossentropy.(decoder_z, x)) / len
+    logp_x_z = -logitbinarycrossentropy(decoder_z, x, agg=sum) / len
```
Comment (CarloLucibello) [inline vision/vae_mnist/vae_mnist.jl:64]: So this was causing the grey images? I would have thought that 
```julia
sum(logitbinarycrossentropy.(decoder_z, x))
```
and
```julia
logitbinarycrossentropy(decoder_z, x, agg=sum)
```
produced the same output...

Comment (CarloLucibello) [reply vision/vae_mnist/vae_mnist.jl:64]: ah now I see the problem was in a misplaced sigmoid

**Pull request merged**